prboom2/src/p_inter.c

```diff
@@ -2298,7 +2307,8 @@ dboolean Heretic_P_GiveWeapon(player_t * player, weapontype_t weapon)
         player->bonuscount += BONUSADD;
         player->weaponowned[weapon] = true;
         P_GiveAmmo(player, wpnlev1info[weapon].ammo, GetWeaponAmmo[weapon]);
-        player->pendingweapon = weapon;
+        if (autoswitch)
+          player->pendingweapon = weapon;
```

Comment:
The condition below is using brackets, so lets follow the style. Ik its not consistent at all, but its better if new code tries to match whats already there.

And maybe you should just extract this `if` into a new `P_AutoSwitchWeapon`

Reply:
Hmmm we could add a `P_AutoSwitchWeapon`, although not all of these cases are `player->pendingweapon = weapon`...

The `play->pendingweapon` is the same, but the result is variable. So doing a `P_AutoSwitchWeapon()` would require running a variable through, which at that point idk if it's worth the hassle.

That is unless you were thinking of replacing all autoswitch cases with a function... although the conditions around each autoswitch seem quite variable, so it doesn't seem like a major boon.

Actually regarding the brackets style, I'm fine with adding brackets. Although I usually tend to avoid using brackets for 1 line conditions, as I usually find it looks cleaner.

Reply:
Ignore my thoughts on `P_AutoSwitchWeapon`... I changed my mind